### PR TITLE
Version 2.0

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
This PR Covers the following items.

- Version 2.0 bump
- Gradle 7 Upgrade
- JDK 8 removal
- Handled removal of mapping types in SQL Plugin.

 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/303
https://github.com/opensearch-project/sql/issues/466
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apply plugin: 'groovy'
+apply plugin: 'java'
+
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
```
Comment (penghuo) [inline buildSrc/build.gradle:9]: does this change is required by all plugin? if so, could you add related issue/pr.

File: buildSrc/src/main/groovy/com/wiredforcode/spawn/KillProcessTask.groovy [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.wiredforcode.gradle.spawn
+
+import org.gradle.api.tasks.TaskAction
+
+class KillProcessTask extends DefaultSpawnTask {
```
Comment (joshuali925) [inline buildSrc/src/main/groovy/com/wiredforcode/spawn/KillProcessTask.groovy:5]: what is this for?

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,8 +1,5 @@
-# Copyright OpenSearch Contributors
-# SPDX-License-Identifier: Apache-2.0
```
Comment (joshuali925) [inline gradle/wrapper/gradle-wrapper.properties:2]: could you keep license header

File: gradlew [hunk 4]
```diff
@@ -1,17 +1,5 @@
...
-#
-# Copyright OpenSearch Contributors
-# SPDX-License-Identifier: Apache-2.0
-#
-# The OpenSearch Contributors require contributions made to
-# this file be licensed under the Apache-2.0 license or a
-# compatible open source license.
-#
-# Modifications Copyright OpenSearch Contributors. See
-# GitHub history for details.
```
Comment (joshuali925) [inline gradlew:12]: could you keep license header, same for gradlew.bat

**Review by penghuo — COMMENTED** [review]
Could you add some comments that why tests in integ-test/doctest are deleted. If it is for deprecate legacy not used test, I would prefer to have a separated PR .

**Review by penghuo — COMMENTED** [review]
Please also help revert change introduce in https://github.com/opensearch-project/sql/pull/483. We could do it separately.

File: buildSrc/build.gradle (hunk 1 above)
Comment (vamsi-amazon) [reply buildSrc/build.gradle:9]: We are using a third party plugin to spawn and kill process in doctest module. The plugin is around 5 years old and there is no active development. As part of Gradle 7 upgrade spawn plugin is throwing errors. As part of fix, imported the plugin code to buildSrc and upgraded to Gradle 7.[The only way for upgrading unmaintained third party plugins]. I couldn't find another well maintained plugin to replace.

Comment (vamsi-amazon) [reply buildSrc/build.gradle:9]: https://github.com/opensearch-project/sql/issues/303.  Will add to the description.

File: gradlew (hunk 4 above)
Comment (vamsi-amazon) [reply gradlew:12]: Will do it. I created gradle wrapper for 7 and it replaced the license header.

File: buildSrc/src/main/groovy/com/wiredforcode/spawn/KillProcessTask.groovy (hunk 2 above)
Comment (vamsi-amazon) [reply buildSrc/src/main/groovy/com/wiredforcode/spawn/KillProcessTask.groovy:5]: https://github.com/opensearch-project/sql/pull/507#discussion_r833802454

Commented here.

**Comment by vamsi-amazon** [comment]
> Could you add some comments that why tests in integ-test/doctest are deleted. If it is for deprecate legacy not used test, I would prefer to have a separated PR .

@penghuo  It was deleted after consulting with @dai-chen. The reason to remove them in this CR is when we are bumping to 2.0, these files are causing build issues with breaking changes from upstream libraries. I felt the better way to handle is to remove them in this CR itself.

**Comment by dai-chen** [comment]
> 

yeah, the `doctest` package is Java code that generates the docs initially. It is not used any more and should be safe to delete.

**Comment by dai-chen** [comment]
> > Could you add some comments that why tests in integ-test/doctest are deleted. If it is for deprecate legacy not used test, I would prefer to have a separated PR .
> 
> @penghuo It was deleted after consulting with @dai-chen. The reason to remove them in this CR is when we are bumping to 2.0, these files are causing build issues with breaking changes from upstream libraries. I felt the better way to handle is to remove them in this CR itself.
>
yeah, the `doctest` package is Java code that generates the docs initially. It is not used any more and should be safe to delete.

File: buildSrc/build.gradle (hunk 1 above)
Comment (penghuo) [reply buildSrc/build.gradle:9]: thanks. could we create an issue to tracking move this dependency later. ideally, we should not depend on it.

**Review by penghuo — APPROVED** [review]
Thanks for the change!

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Pull request merged**